fraction/fraction.go

```diff
@@ -0,0 +1,204 @@
+	switch len(Denominator) {
+	case 0:
+		return fraction{Numerator, 1}
+	case 1:
+		return fraction{Numerator, Denominator[0]}
+	default:
+		panic("improper use of NewFraction, can only have one denominator")
+	}
+}
+
+// GetNumerator - return the Numerator
+func (f fraction) GetNumerator() int64 {
+	return f.Numerator
+}
+
+// GetDenominator - return the Denominator
+func (f fraction) GetDenominator() int64 {
+	return f.Denominator
+}
+
+// RectifySign - make any negative sign exist only in Numerator
+func (f fraction) RectifySign() Fraction {
+	if f.Denominator < 0 {
+		return fraction{-1 * f.Numerator, -1 * f.Denominator}
+	}
+	return f
+}
+
+// Inv - Inverse
+func (f fraction) Inv() Fraction {
+	return fraction{f.Denominator, f.Numerator}
+}
+
+// Simplify - find the greatest common Denominator, divide
+func (f fraction) Simplify() Fraction {
+	gcd := f.Numerator
+	for d := f.Denominator; d != 0; {
+		gcd, d = d, gcd%d
+	}
+	f.Numerator /= gcd
+	f.Denominator /= gcd
+	return f.RectifySign()
+}
+
+// Negative - is the fractior negative
+func (f fraction) Negative() bool {
+	switch {
+	case f.Numerator > 0:
+		if f.Denominator > 0 {
+			return false
+		}
+		return true
+	case f.Numerator < 0:
+		if f.Denominator < 0 {
+			return false
+		}
+		return true
+	}
+	return false
+}
+
+// Positive - is the fraction positive
+func (f fraction) Positive() bool {
+	switch {
+	case f.Numerator > 0:
+		if f.Denominator > 0 {
+			return true
+		}
+		return false
+	case f.Numerator < 0:
+		if f.Denominator < 0 {
+			return true
+		}
+		return false
+	}
+	return false
+}
+
+// Equal - test if two Fractions are equal, does not simplify
+func (f fraction) Equal(f2 Fraction) bool {
+	if f.Numerator == 0 {
+		return f2.GetNumerator() == 0
+	}
+	f1 := f.RectifySign()
+	f2 = f2.RectifySign()
+	return ((f1.GetNumerator() == f2.GetNumerator()) &&
+		(f1.GetDenominator() == f2.GetDenominator()))
+}
+
+// GT - greater than
+func (f fraction) GT(f2 Fraction) bool {
+	return f.Sub(f2).Positive()
+}
+
+// LT - less than
+func (f fraction) LT(f2 Fraction) bool {
+	return f.Sub(f2).Negative()
+}
+
+// Mul - multiply
+func (f fraction) Mul(f2 Fraction) Fraction {
+	return fraction{
+		f.Numerator * f2.GetNumerator(),
+		f.Denominator * f2.GetDenominator(),
+	}.Simplify()
+}
+
+// Div - divide
+func (f fraction) Div(f2 Fraction) Fraction {
+	if f2.GetNumerator() == 0 {
+		panic("fraction divide by zero error!")
+	}
+	return fraction{
+		f.Numerator * f2.GetDenominator(),
+		f.Denominator * f2.GetNumerator(),
+	}.Simplify()
+}
+
+// Add - add without simplication
+func (f fraction) Add(f2 Fraction) Fraction {
+	if f.Denominator == f2.GetDenominator() {
+		return fraction{
+			f.Numerator + f2.GetNumerator(),
+			f.Denominator,
+		}.Simplify()
+	}
+	return fraction{
+		f.Numerator*f2.GetDenominator() + f2.GetNumerator()*f.Denominator,
+		f.Denominator * f2.GetDenominator(),
+	}.Simplify()
+}
+
+// Sub - subtract without simplication
```

Comment:
how's that `without simplication` if you call Simplify in the end?

Reply:
antiquated, now using big.Rat